### PR TITLE
Load now produces the history of the loader called

### LINKED ISSUE #32950: Load does not include optional arguments in script
Load some data into the muon GUI and wait for project recovery to finish
Crash Mantid
When you try to recover you will get the error:

```
ValueError: Invalid value for property InputWorkspace (Workspace) from string "MUSR00062260.nxs_deadtime_table": Workspace "MUSR00062260.nxs_deadtime_table" was not found in the Analysis Data Service
  File "C:\Users\BTR75544\AppData\Roaming\mantidproject\mantid\ordered_recovery.py", line 4, in <module>
    RenameWorkspace(InputWorkspace='MUSR00062260.nxs_deadtime_table', OutputWorkspace='MUSR62260_deadtime MA') # 2021-11-11T16:59:09.792657000 execCount: 5
  File "C:\Users\BTR75544\nightly\MantidNightlyInstall\bin\mantid\simpleapi.py", line 560, in RenameWorkspace
    algm.setProperty(key, val)
```

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**

The history from Load did not include the correct properties, as it would ignore properties for the specific flavour of the load that was used. This caused loading muon data via "Load" to not produce the correct history. 

**To test:**

Open Muon Analysis
Load some data
Grab the history from any of the workspaces and put it into a script
Clear the ADS
Run the script
All of the workspaces should be reproduced. 

Fixes #32950. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by martyngigg — CHANGES_REQUESTED** [review]
Code wise this seems like the right thing but I've tried it in workbench and I seem to get an extra property on the generated history that I think is an output property (`TimeZeroList`)? I don't think that was here before? I've included a screenshot:

Screenshot 2021-11-19 at 13 23 10

File: Framework/API/test/ScriptBuilderTest.h [hunk 1]
```diff
@@ -533,6 +533,38 @@ class ScriptBuilderTest : public CxxTest::TestSuite {
     m_ads.remove("test_output_workspace");
   }
 
+  void test_Build_Load_Uses_Args_From_Correct_Load() {
```
[the remark this answers is not in the document]
Comment (AnthonyLim23) [reply Framework/API/test/ScriptBuilderTest.h:536]: yes, you are right.  I will just need to work out how to handle the file path changing between OS's

**Comment by AnthonyLim23** [comment]
> Code wise this seems like the right thing but I've tried it in workbench and I seem to get an extra property on the generated history that I think is an output property (`TimeZeroList`)? I don't think that was here before? I've included a screenshot:
> 
> Screenshot 2021-11-19 at 13 23 10

TimeZeroList should have present before. It is used for working out the time zero shifts (each detector could have a unique value).

**Comment by martyngigg** [comment]
Thanks for updating the test.

Looking at the loader I can see `TimeZeroList` is an [output property](https://github.com/mantidproject/mantid/blob/442f0c7f0b5a046cfce5b3bd5525de4162323a7c/Framework/DataHandling/src/LoadMuonNexus.cpp#L95) so I don't think it should be in the generated script?

**Comment by AnthonyLim23** [comment]
I think we use it later on in the history. It should be replaced at some point by a table.

**Review by martyngigg — APPROVED** [review]
Ah I see this happens with `LoadMuonNexus` also. That means this essentially works as it did before so I'll approve it.

**Pull request merged**